### ISSUE TITLE
Compiling under Windows - bootstrap.ps1 not compatible with Visual Studios 2019

### ISSUE THREAD

**Issue by kheckwrecker** [vegastrike/Vega-Strike-Engine-Source#906]
When compiling under Windows, using Visual Studios 2019, bootstrap.ps1 fails with:

CMake Error: Could not read presets from D:/VegaStrike-0.9.0/Vega-Strike-Engine-Source-0.9.0/engine: Unrecognized "version" field.

Apparently, this is a known issue with the version of CMake used to compile and VS 2019. 

However, build.ps1 expects to find VS 2019. The work-around is to install both VS 2022 and VS 2019 and run the compile through VS 2022. 

It would be nice, though, to be able to compile with one Visual Studio or the other.

**Comment by royfalk** [comment]
OK. These are actually two issues:
1. VS2019 has a CMake error.
2. build.ps1 expects VS2019.

For the first one, can you provide a more full error log, just in case the lines before and after are also relevant.
For the second one, that should be a simple fix.

**Comment by stephengtuggy** [comment]
When you run the build.ps1 script, you can pass a command-line parameter that tells it which version of Visual Studio you want to use. I think the same may apply to bootstrap.ps1; I forget exactly.

As for the `Unrecognized "version" field` error: I suspect that that has more to do with the version of CMake being used. Can you run `cmake --version` and report back the output?

Thanks! Sorry for all the hassle!

**Comment by kheckwrecker** [comment]
I ran CMake --version for both Visual Studio 2022 and Visual Studio 2019 in case you needed both:

This is the version in D:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/bin/

cmake version 3.29.5-msvc4

And this is the version in C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin

cmake version 3.20.21032501-MSVC_2

**Comment by stephengtuggy** [comment]
Thank you for reporting back. Sure enough, according to https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html , CMakePresets version 3 is only supported by CMake 3.21 and up. So it's probably best to use the Visual Studio 2022 tooling, like you're doing. I didn't realize that VS2019's version of CMake was that old.

**Comment by stephengtuggy** [comment]
If you like, you can try clearing out the build folders again, and then run: `.\script\build.ps1 -Generator VS2022Win64` . That should build everything using the Visual Studio 2022 toolchain. But up to you if you want to go through the whole process of clearing everything out and then building it again.

**Comment by kheckwrecker** [comment]
I deleted the build folders, uninstalled VS2019, redownloaded the latest master, successfully ran bootstrap.ps1, but build initially failed with the following error:

CMake Error at D:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Python3 (missing: Interpreter) (found version "3.11.8")

      Reason given by package:
          Interpreter: Cannot run the interpreter "D:/VS9/build/VS2022Win64-pie-disabled-release/vcpkg_installed/x64-windows-win10/tools/python3/python.exe"

Call Stack (most recent call first):
  D:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  D:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.29/Modules/FindPython/Support.cmake:3867 (find_package_handle_standard_args)
  D:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.29/Modules/FindPython3.cmake:545 (include)
  D:/VS9/build/VS2022Win64-pie-disabled-release/vcpkg_installed/x64-windows-win10/share/python3/vcpkg-cmake-wrapper.cmake:67 (_find_package)
  D:/Program Files/Microsoft Visual Studio/2022/Community/VC/vcpkg/scripts/buildsystems/vcpkg.cmake:813 (include)
  CMakeLists.txt:374 (FIND_PACKAGE)


-- Configuring incomplete, errors occurred!

So, I unset PYTHONHOME in the VS 2022 Developer Shell with the following:

[Environment]::SetEnvironmentVariable("PYTHONHOME", "")

then re-ran  "script/build.ps1 -Generator VS2022Win64" and this time it completed successfully. I was able to run the game, but ran into the previously reported issue #931, so I'll need to manually edit the settings file to get past the main menu.

Long story short, the game compiles successfully using VS 2022 (only).

**Comment by kheckwrecker** [comment]
Was able to get around the screen resolution issue once I found and edited config.json, so the game now runs successfully.

**Comment by stephengtuggy** [comment]
@kheckwrecker Good to know! Thanks for putting in all that effort. That is valuable information.

**Comment by stephengtuggy** [comment]
@kheckwrecker can this ticket be closed then?

**Comment by kheckwrecker** [comment]
Yes, absolutely. The readme needs to be updated, but that's already reported as a separate issue.
